Makefile

```diff
@@ -40,33 +47,35 @@ start:
 dev-server: CONFIG_ENV = development
 dev-server: CALYPSO_ENV = desktop-development
 dev-server: NODE_ENV = development
-dev-server: checks
+dev-server: checks desktop$/config.json
 	@echo "\n\n$(GREEN)+------------------------------------------------+"
 	@echo "|                                                |"
 	@echo "|    Wait for calypso to start the dev server    |"
 	@echo "|       and start the app with \`make dev\`        |"
 	@echo "|                                                |"
 	@echo "+------------------------------------------------+$(RESET)\n\n"
-	
-	@$(MAKE) build-config CONFIG_ENV=$(CONFIG_ENV)
-
 
 	@npx concurrently -k \
 	-n "Calypso,Desktop" \
 	"$(MAKE) calypso-dev NODE_ENV=$(NODE_ENV) CALYPSO_ENV=$(CALYPSO_ENV)" \
-	"wait-on http://localhost:3000 && $(MAKE) desktop-dev NODE_ENV=$(NODE_ENV)" \
+	"wait-on http://localhost:3000 && $(MAKE) build-desktop NODE_ENV=$(NODE_ENV)" \
 
 # Start app in dev mode
 dev: NODE_ENV = development
 dev: DEBUG = desktop:*
 dev: 
 	$(MAKE) start NODE_ENV=$(NODE_ENV) DEBUG=$(DEBUG)
 
-# Build config
-build-config: 
-	@node $(THIS_DIR)/resource/build-scripts/build-config-file.js $(CONFIG_ENV)
+desktop-config$/config-%.json:
+	$(info Config file for environment "$(CONFIG_ENV)" does not exist. Ignoring Environment.)
 
-	@echo "$(CYAN)$(CHECKMARK) Config created$(RESET)"
+.PHONY: desktop$/config.json
```

Comment:
as we discussed in Slack we can refine this further by creating a kind of "lock file" for the environment variables. I don't think it's necessary to do that here in this PR but I'm leaving a note for the future reader that by writing more idiomatic `Makefile` code like is happening here (instead of using a custom JS runner) then we make it easier to refine and optimize our build process.

someone should be able to come around to this line and say, "oh hey, we don't have to regenerate the config files every time, we can add `.env` as a dependent and skip some work" whereas with the JS runner it's number one _hidden from sight_ and number two it _has_ to run every time